### PR TITLE
cli: fix module discovery race

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

Had reports of trouble with the module file failing to compile. Looking into things I believe this is because we're sending multiple concurrent writes the the same file. Something like this might happen:

1. Start writing big content, e.g. `123456`
2. Start writing small content, e.g. `abc`
3. Finish writing big content
4. Finish writing small content

And then you're left with the file content `abc456`

👀 @iamEAP 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by benjdlambert — APPROVED** [review]
That'd do it 😅

File: packages/cli/src/lib/bundler/packageDetection.ts [hunk 1]
```diff
@@ -116,13 +120,16 @@ async function writeDetectedPackagesModule(
     )
     .join(',');
 
-  await fs.writeFile(
-    joinPath(
-      cliPaths.targetRoot,
-      'node_modules',
-      `${DETECTED_MODULES_MODULE_NAME}.js`,
+  writeQueue.clear();
```
Comment (freben) [inline packages/cli/src/lib/bundler/packageDetection.ts:123]: Did you check that this doesn't also immediately detach/"orphan" any ongoing promises?

Could just as well skip this line actually if it's an undefined behavior (the docs don't really say clearly), not sure it'll ever truly matter.

**Review by freben — COMMENTED** [review]
Hmm looks good but what about that clear

File: packages/cli/src/lib/bundler/packageDetection.ts (hunk 1 above)
Comment (Rugvip) [reply packages/cli/src/lib/bundler/packageDetection.ts]: Yep it's not clear so might as well not clear

**Pull request merged**